apps/wizard/app_pages/chart_diff/conflict_resolver.py

```diff
@@ -0,0 +1,172 @@
+            for field in self.config_compare:
+                self._show_field_conflict_resolver(field)
+
+            # Button to resolve all conflicts
+            if st.button(
+                "Resolve conflicts",
+                help="Click to resolve the conflicts and update the chart config.",
+                key="resolve-conflicts-btn",
+                type="primary",
+            ):
+                self.resolve_conflicts(rerun=True)
+
+    def _show_field_conflict_resolver(self, field):
+        with st.container(border=True):
+            # Title & layout
+            st.header(field["key"])
+
+            # Choose option
+            choice = self._choose_env(field)
+
+            # Show the fields values
+            col1, col2 = st.columns(2)
+            with col1:
+                with st.expander("PRODUCTION"):
+                    st.write(field["value1"])
+            with col2:
+                with st.expander("STAGING"):
+                    st.write(field["value2"])
+
+            # Merge editor
+            self._show_merge_editor(field, choice)
+
+    def _choose_env(self, field) -> int:
+        """Choose environment to keep the value from.
+
+        The user can edit the value later on.
+        """
+        # Show option radio buttons
+        value = st.radio(
+            "Choose config from...",
+            options=[1, 2],
+            captions=["Insert production config", "Insert staging config"],
+            format_func=lambda x: ENVIRONMENT_IDS[x],
+            key=f"conflict-radio-{field['key']}",
+            horizontal=True,
+            label_visibility="collapsed",
+        )
+
+        return cast(int, value)
+
+    def _show_merge_editor(self, field, choice):
+        """Edit the content of the field."""
+        is_none = field[f"value{choice}"] is None
+        self.value_resolved[field["key"]] = st.text_area(
+            "Edit config",
+            value=None if is_none else str(field[f"value{choice}"]),
+            placeholder=f"This field is not present in {ENVIRONMENT_IDS[choice]}!" if is_none else "",
+            help="Edit the final config here. When cliking on 'Resolve conflicts', this value will be used to update the chart config.",
+            disabled=is_none,
+        )
+
+    def resolve_conflicts(self, rerun: bool = False):
+        """Gather all resolved conflicts and update chart config in staging."""
+        with st.spinner("Updating chart on staging..."):
+            # Consolidate changes
+            config = deepcopy(self.diff.source_chart.config)
+            for field_key, field_resolution in self.value_resolved.items():
+                if self.value_resolved[field_key] is None:
+                    config.pop(field_key, None)
+                else:
+                    config[field_key] = as_valid_json(field_resolution)
+
+            # Push to staging
+            api = AdminAPI(SOURCE.engine, grapher_user_id=1)
```

Comment:
(In my other PR I'm assuming that everything on the staging server is done by Admin user.)

Reply:
should i change to use the defaults? I am not sure what happens if the user is not using STAGING, and then the chart gets modified by X user id. This would prevent the tool to work properly.

But at the same time these tools are not currently working for local dev.

Reply:
It's fine. I'll let everyone know to change it.